src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp

```diff
@@ -298,12 +299,23 @@ namespace OpenMS
     newProtIDRun.setSearchEngine(cmap.getProteinIdentifications()[0].getSearchEngine());
     newProtIDRun.setSearchEngineVersion(cmap.getProteinIdentifications()[0].getSearchEngineVersion());
     newProtIDRun.setSearchParameters(cmap.getProteinIdentifications()[0].getSearchParameters());
-    //TODO based on old IDRuns or based on consensusHeaders?
-    //vector<String> mergedOriginFiles = cmap.getProteinIdentifications()...
+    String old_inference_engine = cmap.getProteinIdentifications()[0].getInferenceEngine();
+    if (!old_inference_engine.empty())
+    {
+      OPENMS_LOG_WARN << "Inference was already performed on the runs in this ConsensusXML."
+                         " Merging their proteins, will invalidate correctness of the inference."
+                         " You should redo it.\n";
+      // deliberately do not take over old inference settings.
+    }
+
+    //we do it based on the IDRuns since ID Runs maybe different from quantification in e.g. TMT
     vector<String> mergedOriginFiles{};
-    for (const auto& chead : cmap.getColumnHeaders())
+    //set<String> mergedOriginFiles{};
```

Comment:
remove commented out?

Reply:
I am actually not sure, what to take here.